packages/shell-api/src/shard.ts

```diff
@@ -72,12 +79,91 @@ export default class Shard extends ShellApiClass {
       cmd.unique = unique;
     }
     const orderedCmd = options !== undefined ? { ...cmd, ...options } : cmd;
-    return this._mongo._serviceProvider.runCommandWithCheck(ADMIN_DB, orderedCmd);
+    try {
+      return await this._mongo._serviceProvider.runCommandWithCheck(ADMIN_DB, orderedCmd);
+    } catch (error) {
+      if (error.codeName === 'CommandNotFound') {
+        error.message = `${error.message}. Are you connected to mongos?`;
+      }
+      throw error;
+    }
   }
 
   @returnsPromise
   async status(verbose = false): Promise<any> {
+    await getConfigDB(this._mongo);
     const result = await getPrintableShardStatus(this._mongo, verbose);
     return new CommandResult('StatsResult', result);
   }
+
+  @returnsPromise
+  async addShard(url: string): Promise<any> {
+    assertArgsDefined(url);
+    await getConfigDB(this._mongo);
+    this._emitShardApiCall('addShard', { url });
+    return this._mongo._serviceProvider.runCommandWithCheck(ADMIN_DB, {
+      addShard: url
+    });
+  }
+
+  @returnsPromise
+  @serverVersions(['3.4.0', ServerVersions.latest])
+  async addShardToZone(shard: string, zone: string): Promise<any> {
+    assertArgsDefined(shard, zone);
+    this._emitShardApiCall('addShardToZone', { shard, zone });
+    await getConfigDB(this._mongo); // error if not connected to mongos
+    return this._mongo._serviceProvider.runCommandWithCheck(ADMIN_DB, {
+      addShardToZone: shard,
+      zone: zone
+    });
+  }
+
+  @returnsPromise
+  @serverVersions(['3.4.0', ServerVersions.latest])
+  async addShardTag(shard: string, tag: string): Promise<any> {
+    try {
+      return await this.addShardToZone(shard, tag);
+    } catch (error) {
+      if (error.codeName === 'CommandNotFound') {
+        error.message = `${error.message}. Are you connected to version > 3.4?`;
+      }
+      throw error;
```

Comment:
Not necessarily something for this PR, but is this something that would make sense to implement inside of `@serverVersions`?

Reply:
This one is a bit of an odd case because the old shell changed the behavior of this helper in 3.4, but we only support 4.0+ in the new shell so we don't want to handle the case that the user is calling this helper on an old server version. 

If we added it to the decorator, it would apply for all methods and I don't think we want to add this hint to every time we try a command that doesn't exist when we decorate the method with a specific version (since the decorations, for now, are just for autocomplete). I think the error message could be a bit clearer, since what I'm really trying to say is 'are you expecting this to work the way the old shell did, on your old version of the server, because it won't work because we don't support that'.

Reply:
Okay, makes sense, I didn’t have that context :)

Reply:
I've updated the message to make it clearer what's going on, I think its better now